dashboards/clicks_and_impressions.py

```diff
@@ -0,0 +1,92 @@
+import streamlit as st
+import matplotlib.pyplot as plt
+
+from utils.duckdb import get_dbcur
+
+st.title("Clicks & Impressions Dashboard")
+
+cur = get_dbcur()
+
+query = """
+WITH ranked_events AS (
+    SELECT 
+        ed.website_event_id,
+        es.id AS mentor_id,
+        ANY_VALUE(es.name) AS mentor_name,
+        SUM(CASE WHEN we.event_name = 'Click' THEN 1 ELSE 0 END) AS click_count,
+        SUM(CASE WHEN we.event_name = 'Impression' THEN 1 ELSE 0 END) AS impression_count
```

Comment:
An additional condition is required to remove tracked events on development instances (see https://github.com/AdvisorySG/mentorship-streamlit/pull/46#pullrequestreview-2522572346 for details). Perhaps add something like `SUM(CASE WHEN ed.data_key = 'env' AND ed.string_value = 'production' THEN 1 ELSE 0 END) AS is_production`, then add a filter later on to select only columns with `is_production = 1`?

Reply:
Actually, such a filter won't work - I see that the join criteria at [L25](https://github.com/AdvisorySG/mentorship-streamlit/pull/46/files#diff-70d43095056fd0c892b4c6ca5c1c4f1882de4af2645b15300f724d302f549120R25) means all of the rows matching `ed.data_key = 'id'` won't be included in this final table. From my understanding, we probably need to do a subquery to retain only the production events in `ed`.